src/utils/openai.ts

```diff
@@ -97,46 +97,113 @@ const sanitizeMessage = (message: string) => message.trim().replace(/[\n\r]/g, '
 
 const deduplicateMessages = (array: string[]) => Array.from(new Set(array));
 
-const getPrompt = (locale: string, diff: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the following diff without prefacing it with anything, the response must be in the language ${locale}:\n${diff}`;
+const getBasePrompt = (locale: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the diff that I provide you without prefacing it with anything, the response must be in the language ${locale}`;
+
+const getCommitMessageFormatPrompt = (useConventionalCommits: boolean) => {
+	const commitTitleParts = [];
+
+	if (useConventionalCommits) {
+		commitTitleParts.push('<conventional commits type>(<optional scope of the change>):');
+	}
+
+	commitTitleParts.push('<commit message>');
+
+	return commitTitleParts.join(' ');
+};
+
+const getExtraContextForConventionalCommits = () => {
+	// Based on https://medium.com/neudesic-innovation/conventional-commits-a-better-way-78d6785c2e08
+	const conventionalCommitTypes: Record<string, string> = {
+		/*
+			Commented out feat: fix: change: :refactor because they are too common and
+			will cause the model to generate them too often.
+		*/
+		// feat: 'The commit implements a new feature for the application.',
+		// fix: 'The commit fixes a defect in the application.',
+		// change: 'changes the implementation of an existing feature',
+		// refactor: 'code refactoring',
+		build: 'alters the build system or external dependencies of the product',
+		chore: 'includes a technical or preventative maintenance task',
+		ci: 'continuous integration or continuous delivery scripts or configuration files',
+		deprecate: 'deprecates existing functionality',
+		docs: 'changes to README files and markdown (*.md) files',
+		perf: 'improve the performance of algorithms or general execution',
+		remove: 'removes a feature or dependency',
+		revert: 'reverts one or more commits',
+		security: 'improves security',
+		style: 'updates or reformats the style of the source code',
+		test: 'changes to the suite of automated tests',
+	};
+
+	let conventionalCommitDescription = '';
+	// eslint-disable-next-line guard-for-in
+	for (const key in conventionalCommitTypes) {
+		const value = conventionalCommitTypes[key];
+		conventionalCommitDescription += `${key}: ${value}\n`;
+	}
+
+	return `Choose the primary used conventional commit type from the list below based on the git diff:\n${conventionalCommitDescription}`;
+};
 
 const model = 'gpt-3.5-turbo';
+// TODO: update chat completion to using the openai SDK
+const encoder = encodingForModel('text-davinci-003');
```

Comment:
```suggestion
const encoder = encodingForModel(model);
```

But also, I previously removed the variable because it's only used once, and it's initialized in memory even before we know that the project is ready to commit.